packages/stable/src/api/annotations/annotationsApi.ts

```diff
@@ -140,4 +143,32 @@ export class AnnotationsAPI extends BaseResourceAPI<AnnotationModel> {
   public update = (changes: AnnotationChangeById[]) => {
     return this.updateEndpoint(changes);
   };
+
+  /**
+   * [Reverse lookup resources](TODO: use API Doc link when ready)
+   * const assetQueryData = {
+   *   limit: -1,
+   *   filter: {
+   *     annotatedResourceType: 'file' as const,
+   *     annotationType: 'images.AssetLink',
+   *     data: {
+   *       { id: 123 }
+   *     }
+   *   }
+   * };
+   *
+   * const resourceIdsResponse = this._client.annotations.reverseLookup(assetQuerydata);
+   */
+  public reverseLookup = (
+    filter: AnnotationReverseLookupRequest
+  ): CursorAndAsyncIterator<AnnotationsAssetRef> => {
```

Comment:
![](https://i.kym-cdn.com/entries/icons/original/000/032/744/maneuver.jpg)

This breaks with the convention in the rest of the SDK, so I had to use a lot of lower-level components.. But I think it's done now

Reply:
ah, right! If you decide that users can get by without that field, i won't require that you add it. it's certainly difficult when you also want to keep the cursor and async iterator

It seems the annotatedResourceType is specified in the request structure, so perhaps it's just a helpful field, but not strictly important?

Reply:
Yes, to be honest I don't understand why that field was added in the response, since it will, to my knowledge, always be the same as the one in the request :sweat: So yes, maybe I'll just revert :thinking: Makes it much easier to maintain